### ISSUE TITLE
Error 0x80370102 after wsl --install and reboot

### ISSUE THREAD

**Issue by HunterZ** [microsoft/WSL#7902]
### Version

10.0.22000.376

### WSL Version

- [X] WSL 2
- [X] WSL 1

### Kernel Version

5.10.60.1

### Distro Version

N/A

### Other Software

N/A

### Repro Steps

1. In Windows Features, disable the following:
- Hyper-V
- Virtual Machine Platform
- Windows Hypervisor Platform
- Windows Subsystem for Linux
2. Reboot
3. Open console, and perform the following:
- run `wsl --install`
- observe that Virtual Machine Platform and Windows Subststem for Linux features are enabled
- observe that Ubuntu image is downloaded
- observe reboot request
4. Reboot

### Expected Behavior

`wsl --install` should result in a working WSL setup when prerequisites are met. If something needs to be enabled or installed, then it should do that, and not leave me with a broken install.

### Actual Behavior

The following appears in the console:
```Installing, this may take a few minutes...
WslRegisterDistribution failed with error: 0x80370102
Error: 0x80370102 The virtual machine could not be started because a required feature is not installed.

Press any key to continue...

[process exited with code 1]
```

### Diagnostic Logs

[diagnostics.zip](https://github.com/microsoft/WSL/files/7831945/diagnostics.zip)
I've attached a zip containing a wsl.wprp event log, a redacted systeminfo output, and a filtered PowerShell Get-ComputerInfo output.

**Comment by HunterZ** [comment]
[coreinfo64.zip](https://github.com/microsoft/WSL/files/7831950/coreinfo64.zip)
Here is a coreinfo64 tool output as well.

**Comment by OneBlue** [comment]
Based on the logs you shared, I can see:

```
HyperVisorPresent                                 : False
``` 

Which is probably what's causing this error.

I wonder if the hypervisor is disabled in your boot configuration. Can you share the output of `bcdedit /enum` ?

**Comment by HunterZ** [comment]
`hypervisorlaunchtype    Off`

This should have been enabled by `wsl --install` if it is a prerequisite. This should have been mentioned in the WSL documentation if it is a prerequisite.

**Help Needed: What specific command should I run?**

I did try re-enabling this in the past, so this won't be the end of this issue report, but it's currently disabled because VirtualBox will NOT run with it enabled, regardless of what anyone says.

**Comment by OneBlue** [comment]
Thanks @HunterZ. [Added a section on hypervisorlaunchtype in the documentation](https://github.com/MicrosoftDocs/WSL/pull/1418).

In your case, running `bcdedit /set {current} hypervisorlaunchtype Auto` should make hyper-V available after a reboot.

**Comment by HunterZ** [comment]
Ran command, rebooted, and now it reports `hypervisorlaunchtype    Auto`

...and somehow `wsl --install -d Ubuntu` actually worked this time. I went through all this before, so I fully expected it to fail this time too. VirtualBox is even working at the same time somehow, which also broke before.
```
C:\Users\bensh>wsl --list --verbose
  NAME      STATE           VERSION
* Ubuntu    Running         2
```

I guess I'll take the win and close the issue, but it's a little unsatisfying; I suspect there's some needle-threading that I was failing to navigate before but somehow managed to get right this time.

Additional notes for reference:
* Hyper-V and Windows Hypervisor Platform are still disabled in Windows Features, so they are definitely NOT required.
* `systeminfo` now detects a hypervisor.
* `coreinfo` reports hypervisor present and blocking Intel hardware virtualization (confirmed by VirtualBox reporting lack of nested paging support).
